### ISSUE TITLE
cache bug

### ISSUE THREAD

**Issue by einsqing** [gilt/koa-hbs#60]
```js
koa.use(convert(hbs.middleware({
    extname: '.html',
    viewPath: ['./www/admin', './www/home'],
    layoutsPath: './www',
    partialsPath: ['./www/admin', './www/home'],
    disableCache: true
})));

koa.use(async(ctx, next) => {
    const render = ctx.render;
    ctx.render = async function _convertedRender() {
        return co.call(ctx, render.apply(ctx, arguments))
    }

    await next();
});

###render###

 if(ctx.query.type){
       ctx.state = {
             layout: 'admin/override'
       };
}

await ctx.render('index_index');
```

If I open the cache, view rendering not to come out, when I closed the cache, the view can render.

**Comment by shellscape** [comment]
I'm not sure I understand. could you rephrase?

**Comment by einsqing** [comment]
![http://www.wemallshop.com/hbs.gif]()

**Comment by einsqing** [comment]
I recorded a GIF
http://www.wemallshop.com/hbs.gif

**Comment by einsqing** [comment]
![Uploading 497D7471-FBC4-48DB-9527-1F7CFBA7B8AF.png…]()
Actually return data, but this. The body has no output to the browser

**Comment by shellscape** [comment]
so that looks like a koa2 issue. support for that is hacked together with koa-convert and some middleware trickery, but it's not built in. if you find a solution, I'd be happy to add it to the readme. but this might have to wait until Koa2 support is official.

**Comment by einsqing** [comment]
After using async/await rewrite koa-hbs, I found this not koa or koa-convert bug, this is my own writing async function called dynamic, not await.Contribute a koa-hbs for koa2 https://github.com/einsqing/koa2-hbs
